src/status_im/contexts/onboarding/events.cljs

```diff
@@ -1,31 +1,192 @@
+
+(defn biometrics-setup-start
+  [_ [{:keys [enable-biometrics? syncing?]}]]
+  {:fx (cond-> []
+         enable-biometrics?
+         (conj [:dispatch
+                [:onboarding/enable-biometrics
+                 {:on-done [:onboarding/biometrics-setup-done
+                            {:on-success [:onboarding/notifications-setup
+                                          {:biometrics?           true
+                                           :biometrics-supported? true
+                                           :onboarding?           true
+                                           :syncing?              syncing?}]
+                             :on-fail    [:onboarding/biometrics-fail]}]}]])
+
+         (not enable-biometrics?)
+         (conj [:dispatch
+                [:onboarding/notifications-setup
+                 {:biometrics?           false
+                  :biometrics-supported? true
+                  :onboarding?           true
+                  :syncing?              syncing?}]]))})
+
+(rf/reg-event-fx :onboarding/biometrics-setup-start biometrics-setup-start)
+
+(defn biometrics-setup-done
+  [{:keys [db]} [{:keys [on-success on-fail]} {:keys [error]}]]
+  {:db (assoc-in db [:onboarding/profile :auth-method] constants/auth-method-biometric)
+   :fx (cond-> []
+         (some? error)
+         (conj [:dispatch (conj on-fail error)])
+         :else
+         (conj [:dispatch on-success]))})
+
+(rf/reg-event-fx :onboarding/biometrics-setup-done biometrics-setup-done)
+
 (rf/reg-event-fx
- :onboarding/finish-onboarding
- (fn [_ [notifications-enabled?]]
-   {:fx [(when notifications-enabled?
-           [:dispatch [:push-notifications/switch {:notifications-enabled true}]])
-         [:dispatch [:shell/change-tab shell.constants/default-selected-stack]]
-         [:dispatch [:update-theme-and-init-root :screen/shell-stack]]
-         [:dispatch [:profile/toggle-testnet-mode-banner]]
-         [:dispatch [:universal-links/process-stored-event]]]}))
+ :onboarding/biometrics-fail
+ (fn [_ [error]]
+   {:dispatch [:biometric/show-message (ex-cause error)]}))
 
-(rf/defn enable-biometrics
-  {:events [:onboarding/enable-biometrics]}
-  [_]
+(defn enable-biometrics
+  [_ [{:keys [on-done]}]]
   {:fx [[:dispatch
          [:biometric/authenticate
-          {:on-success #(rf/dispatch [:onboarding/biometrics-done])
-           :on-fail    #(rf/dispatch [:onboarding/biometrics-fail %])}]]]})
+          {:on-success #(rf/dispatch on-done)
+           :on-fail    #(rf/dispatch (conj on-done %))}]]]})
+
+(rf/reg-event-fx :onboarding/enable-biometrics enable-biometrics)
+
+;;; -------
+
+(defn inject-local-profile-storage
+  [context]
+  (let [db            (interceptor/get-coeffect context :db)
+        key-uid       (get-in db [:profile/profile :key-uid])
+        local-profile (mmkv/get-object key-uid)]
+    (assoc-in context
+     [:coeffects :local-profile-storage]
+     local-profile)))
```

Comment:
Here's an implementation of a Re-Frame interceptor that synchronously accesses the `mmkv` key-value store, and it retrieves some metadata about a user profile via the key-uid.